src/odemis/gui/cont/tabs/sparc2_align_tab.py

```diff
@@ -2037,6 +2054,26 @@ def _on_spec_ded_aligner_init_move(self, f: Future) -> None:
         except Exception as ex:
             logging.error("spec-ded-aligner initialisation move failed: %s", ex)
 
+    def _on_spec_ded_aligner_pos(self, pos):
+        """
+        Called when the spec-ded-aligner is moved (and the tunnel align mode is active),
+          for updating the MD_FAV_POS_ACTIVE
+        """
+        if self.tab_data_model.main.tab.value != self:
+            # Should never happen, but for safety, we double check
+            logging.warning("Received active spec-ded-aligner position while outside of alignment tab")
+            return
+
+        # Update the axis which are supposed to be recorded: the ones already in MD_FAV_POS_ACTIVE
+        aligner = self.tab_data_model.main.spec_ded_aligner
+        fib_fav_pos = aligner.getMetadata().get(model.MD_FAV_POS_ACTIVE, {})
```

Comment:
From the looks of it, there is now a possibility that `fib_fav_pos` is an empty dict. In this case the position is never set. Is that desired?

Reply:
Yes, correct. This allows to define which position is recorded by Odemis by specifying it in the microscope file. The current "spec-ded-aligner" has actually 3 axes. But we only want to save the Z position. By specifying only the Z pos in FAV_POS_ACTIVE in the microscope file, we can force Odemis to only save that position.